src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -178,18 +188,18 @@ public void UpdateTrackedDocument(Uri uri, SourceText newSourceText)
         // Find the matching document from the LSP solutions.
         foreach (var (lspSolution, isForked) in lspSolutions)
         {
-            var documents = lspSolution.GetDocuments(uri);
-            if (documents.Any())
+            var additionalDocuments = lspSolution.GetTextDocuments<T>(uri);
+            if (additionalDocuments.Any())
             {
-                var document = documents.FindDocumentInProjectContext(textDocumentIdentifier);
+                var document = additionalDocuments.FindTextDocumentInProjectContext(textDocumentIdentifier);
 
                 // Record metadata on how we got this document.
                 var workspaceKind = document.Project.Solution.Workspace.Kind;
                 _requestTelemetryLogger.UpdateFindDocumentTelemetryData(success: true, workspaceKind);
                 _requestTelemetryLogger.UpdateUsedForkedSolutionCounter(isForked);
                 _logger.TraceInformation($"{document.FilePath} found in workspace {workspaceKind}");
 
-                return document;
+                return document as T;
```

Comment:
FindTextDocumentInProjectContext returns a non-nullable value, so this should just be a cast